calico_upgrade/pkg/commands/dryrun.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/docopt/docopt-go"
+
+	"github.com/projectcalico/calico/calico_upgrade/pkg/clients"
+	"github.com/projectcalico/calico/calico_upgrade/pkg/constants"
+	"github.com/projectcalico/calico/calico_upgrade/pkg/migrate"
+)
+
+func DryRun(args []string) {
+	doc := constants.DatastoreIntro + `Usage:
+  calico-upgrade dry-run
+      [--apiconfigv3=<V3_APICONFIG>]
+      [--apiconfigv1=<V1_APICONFIG>]
+      [--output-dir=<OUTPUTDIR>]
+      [--ignore-v3-data]
+
+Example:
+  calico-upgrade dry-run --apiconfigv3=/path/to/v3/config --apiconfigv1=/path/to/v1/config
+
+Options:
+  -h --help                    Show this screen.
+  --apiconfigv3=<V3_APICONFIG> Path to the file containing connection
+                               configuration in YAML or JSON format for
+                               the Calico v1 API.
+                               [default: ` + constants.DefaultConfigPathV3 + `]
+  --apiconfigv1=<V1_APICONFIG> Path to the file containing connection
+                               configuration in YAML or JSON format for
+                               the Calico v3 API.
+                               [default: ` + constants.DefaultConfigPathV1 + `]
+  --output-dir=<OUTPUTDIR>     Directory to store the data migration reports in.
```

Comment:
Whereas I had thought of this tool as taking v1 input and returning v3 output, now we have "output" which is used for reports. I would suggest (although this might not be possible) renaming this flag to `--report-dir=<REPORTDIR>` instead to avoid this issue.